### ISSUE TITLE
IOperatoin does not seem to have a way to specify the ref-kind for a return statement.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#26310]
For example IConditionalOperatoin has IsRef, but IReturnOperation has no information here.  I would expect either an IsRef property, or a RefKind property (since you could have ```ref readonly``` method, and a ```return ref``` statement will give you a ```ref readonly``` value.).  The latter would be preferred to me since it would mean you wouldn't have to go check the containing method to see if it was ```ref``` or ```ref readonly```.

**Comment by CyrusNajmabadi** [comment]
Tagging @heejaechang @mavasani was this an intentional omission?  Or was it accidental? Is there some other non-syntactic way to get this info?

Thanks!

**Comment by mavasani** [comment]
Also tagging @alekseyts

**Comment by CyrusNajmabadi** [comment]
Note: this was found making an actual feature.  The need here is not speculative.  Thanks!

**Comment by AlekseyTs** [comment]
We actually looked at this and decided that it is not needed for IReturnOperation. It should be enough to take a look at the signature of the containing method.

**Comment by CyrusNajmabadi** [comment]
Is there a recommended way to get the containing method?  I can't see anything on IOperation that gives you that.  Without that, ti's a non-trivial walk to try to find hte containing method (with all the varying syntactic forms across the languages).  If there is an easy way to do that, great!  Otherwise, i think we'd either need an extension for that function, or we'd need 'IReturnOperation' to expose this. 

Thoughts @AlekseyTs / @mavasani ?

**Comment by mavasani** [comment]
> Is there a recommended way to get the containing method?

For analyzers, we should have this symbol available on the containing analysis context. However, I don't think there is an easy way to get this for non-analyzer context. I agree with @CyrusNajmabadi that we should expose this information in some way so the client doesn't have to go back to syntax. @AlekseyTs, thoughts?

**Comment by CyrusNajmabadi** [comment]
>For analyzers, we should have this symbol available on the containing analysis context.

Will that be the outermost method?  Or the innermost?  Just wanting to wrap my head around that.  

But yes, there are times when this won't be used in analyzers.  Even recently we've been getting recommendations to just call "semanticModel.GetOperation" on syntax nodes because it can get us the information we need more easily than using the rest of the semantic model API.  So this would def be a gap there. 

Thanks!

**Comment by mavasani** [comment]
> Will that be the outermost method? Or the innermost? Just wanting to wrap my head around that.

That is correct - it will have the outermost method, so probably not the symbol you are looking for if you are within a lambda/local function.

**Comment by CyrusNajmabadi** [comment]
> That is correct - it will have the outermost method, so probably not the symbol you are looking for if you are within a lambda/local function.

Gotcha.  Then i think we def need some answer here.  Even if i'm in an analyzer, trying to figure this out is non-trivial.  Thanks!

**Comment by sharwell** [comment]
> We actually looked at this and decided that it is not needed for IReturnOperation. It should be enough to take a look at the signature of the containing method.

I've been working on this for a couple hours and have not been able to figure out how to get this information.

**Comment by sharwell** [comment]
Oh wow, adding the property to `IReturnOperation` was way more straightforward. 👍
